qa/util/src/main/java/io/camunda/zeebe/qa/util/cluster/TestCluster.java

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.qa.util.cluster;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import io.atomix.cluster.MemberId;
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.ZeebeClientBuilder;
+import io.camunda.zeebe.test.util.asserts.TopologyAssert;
+import io.camunda.zeebe.util.CloseableSilently;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+import org.awaitility.Awaitility;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A convenience class representing a one or more Spring applications that form a Zeebe cluster.
+ *
+ * <p>It's recommended to use the {@link TestClusterBuilder} to build one.
+ *
+ * <p>As the cluster is not started automatically, the nodes can still be modified/configured
+ * beforehand. Be aware however that the replication factor and the partitions count cannot be
+ * modified: if you configure different values directly on your brokers, then you may run into
+ * issues. Keep in mind as well that the gateways and brokers should be treated as immutable
+ * collections.
+ *
+ * <p>Example usage:
+ *
+ * <pre>{@code
+ * final class MyClusteredTest {
+ *   private SpringCluster cluster;
+ *
+ *   &#64;BeforeEach
```

Comment:
:upside_down_face: that looks like a mistake. I checked and it looks like an "@" is accepted in a code block. Same in a couple places, I think we can just replace "&#64;" with "@".

Reply:
It might be a left over from copying the `zeebe-test-container` examples, and I was on Java 8 for quite a while. I vaguely recall you couldn't just put `@`, but :shrug: